### PR TITLE
WIP: Backend feature comment

### LINKED ISSUE #241: Backend feature comments
Users - basic, trader - may comment on specific trading equipments, events etc.

### PR DESCRIPTION
**Pull request by gurkandemir**
closes #241

### CODE REVIEW

File: app/backend/models/comment.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    require: true
+  },
+
+  text: {
+    type: String,
+    require: true
+  },
+
+  date:{
+    type: String,
```
Comment (bahadirh) [inline app/backend/models/comment.js]: We can use the `Date` type here instead.

File: app/backend/routes/trading-eq.js [hunk 2]
```diff
@@ -6,12 +6,13 @@ const {isAuthenticated} = require('../controllers/auth')
...
 const { TradingEquipment } = require('../models/trading-eq')
 const { CurrentTradingEquipment } = require('../models/current-trading-eq')
+const { Comment } = require('../models/comment')
 
 /*
   Get endpoint for information of specific trading equipment.
   Check controller function for more detail
 */
-router.get('/:code', [modelBinder(TradingEquipment, 'TradingEquipment'), modelBinder(TradingEquipmentFollow, 'TradingEquipmentFollow'), modelBinder(CurrentTradingEquipment, 'CurrentTradingEquipment')], tradingEquipmentController.getTradingEquipment)
+router.get('/:code', [modelBinder(TradingEquipment, 'TradingEquipment'), modelBinder(TradingEquipmentFollow, 'TradingEquipmentFollow'), modelBinder(CurrentTradingEquipment, 'CurrentTradingEquipment'), modelBinder(Comment, 'Comment')], tradingEquipmentController.getTradingEquipment)
```
Comment (bahadirh) [inline app/backend/routes/trading-eq.js]: We can merge the `backend-dev` branch into that one and use multipleModelBinder helper here.

File: app/backend/controllers/comments.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
+  Post method for comments.
+  It saves comment to database.
+*/
+module.exports.postComment = async (request, response) => {
+  let Comment = request.models['Comment']
+
+  if(!request.body.related || !request.body.text){
```
Comment (bahadirh) [inline app/backend/controllers/comments.js]: The `validateBody` middleware may be used here to check the request body.

File: app/backend/models/comment.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const {mongoose} =  require('../db');  // The mongodb connector library
+
+let Comment = mongoose.model('Comment', {
+
+  userId: {
+    type: String,
+    require: true
+  },
+
+  related: {
```
Comment (bahadirh) [inline app/backend/models/comment.js]: We may need to discuss how to establish a relationship between a comment and the entity it's about. Only keeping the record of the entity the comment was made on seems inefficient, as that would make it necessary to check many collections for the given object id, which is redundant.

**Comment by gurkandemir** [comment]
Body validator is not workink right now.

related #244

**Pull request merged**